diff --git a/CMakeLists.txt b/CMakeLists.txt
index 650d8f75a..5eecd9b92 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -172,7 +172,9 @@ if(ECM_FOUND)
     if(ECM_VERSION VERSION_GREATER "1.6.0")
         add_subdirectory(images)
         install(FILES org.kde.gcompris.appdata.xml DESTINATION ${KDE_INSTALL_METAINFODIR})
+        install(FILES org.kde.gcompris-teachers.appdata.xml DESTINATION ${KDE_INSTALL_METAINFODIR})
         install(FILES org.kde.gcompris.desktop DESTINATION ${KDE_INSTALL_APPDIR})
+        install(FILES org.kde.gcompris-teachers.desktop DESTINATION ${KDE_INSTALL_APPDIR})
     else()
         message(STATUS "ECM_VERSION is ${ECM_VERSION}, icons and desktop files won't be installed.")
     endif()
diff --git a/images/CMakeLists.txt b/images/CMakeLists.txt
index 23e174e11..d9e672024 100644
--- a/images/CMakeLists.txt
+++ b/images/CMakeLists.txt
@@ -8,6 +8,8 @@ include(ECMInstallIcons)
 ecm_install_icons(ICONS
     256-apps-gcompris-qt.png
     sc-apps-gcompris-qt.svg
+    256-apps-gcompris-teachers.png
+    sc-apps-gcompris-teachers.svg
     DESTINATION ${KDE_INSTALL_ICONDIR}
     THEME hicolor
 )
